File: 2 - Regression/2c - Polynomial Regression/polynimial_regression.py
#	Polynomial Regression 

#Importing the libraries
import numpy as np 
import matplotlib.pyplot as plt
import pandas as pd 

#Import the dataset
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[: , 1:-1].values
y = dataset.iloc[: , -1].values

#Fitting Polynomial Regression to the Dataset
from sklearn.preprocessing import PolynomialFeatures
poly_regressor = PolynomialFeatures(degree=4)
X_poly = poly_regressor.fit_transform(X)
poly_regressor.fit(X_poly,y)

from sklearn.linear_model import LinearRegression
lin_regressor = LinearRegression()
lin_regressor.fit(X_poly, y)

#Predicting test set results
y_pred = lin_regressor.predict(X_poly)
 
# Visualising Polynomial Regression results (for higher resolution and smoother curve)
X_grid = np.arange(min(X), max(X), 0.1)
X_grid = X_grid.reshape(len(X_grid),1)
plt.scatter(X,y,color='red')
plt.plot(X_grid,lin_regressor.predict(poly_regressor.fit_transform(X_grid)),color = 'blue')
plt.title('Truth or Bluff')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()


#Predicting a new result
lin_regressor.predict(poly_regressor.fit_transform([[6.5]]))

File: 2 - Regression/2c - Polynomial Regression/polynimial_regression.R
# Polynomial Regression

#Import the dataset
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]

# Fitting Polynomial Regression to the dataset
    #  # Manual method
    # dataset$Level2 = dataset$Level^2
    # dataset$Level3 = dataset$Level^3
    # dataset$Level4 = dataset$Level^4
    # 
    # regressor = lm(formula = Salary ~ . ,
    #                data = dataset)

 # Automatic Method
regressor = lm(formula = Salary ~ poly(Level,4) ,
                data = dataset)

summary(regressor)

# Predicting Test Set results
y_pred = predict(regressor, newdata = dataset)

# Visualising Polynomial Regression results
library(ggplot2)
ggplot() + 
    geom_point(aes(x = dataset$Level, y = dataset$Salary), 
               color = 'red') +
    geom_line(aes(x = dataset$Level, y = predict(regressor, newdata = dataset)),
              color = 'blue') +
    ggtitle('Truth or Bluff') +
    xlab('Position Label') +
    ylab('Salary')


# Visualising the Polynomial Regression Model results (for higher resolution and smoother curve)
library(ggplot2)
x_grid = seq(min(dataset$Level), max(dataset$Level), 0.1)
ggplot() +
    geom_point(aes(x = dataset$Level, y = dataset$Salary),
               colour = 'red') +
    geom_line(aes(x = x_grid, y = predict(regressor, newdata = data.frame(Level = x_grid))),
              colour = 'blue') +
    ggtitle('Truth or Bluff (Polynomial Regression Model)') +
    xlab('Level') +
    ylab('Salary')

# Predicting a new result
x = 6.5 
y_new_pred = predict(regressor, newdata = data.frame(Level = x))
